lib/services/server/inMemoryDeviceRegistry.js

```diff
@@ -147,13 +150,58 @@ function list(callback) {
     callback(null, result);
 }
 
+
+/**
+ * If Lifetime Resource exists, then the registration SHOULD be removed by the Server if a new registration or update
+ * is not received within this lifetime.
+ *
+ * @param {Object} lifetimeCheckInterval           Minimum interval between lifetime checks in ms
+ */
+function checkLifetime(lifetimeCheckInterval) {
+    checkLifetimeInterval = setInterval(function(){
+        list(function(error, deviceList){
+            if (!error && deviceList) {
+                deviceList.forEach(function(device){
+                    if (device.lifetime &&
+                        new Date() - new Date(device.creationDate) > Number(device.lifetime) * 1000) {
+                        unregister(device.id, function(err, obj){
+                            if (err) {
+                                logger.debug(context,
+                                    'Lifetime unregistration for device [%s] ended up in error [%s] with code [%s]',
+                                    obj.name, err.name, err.code);
+                            }
+                            else {
+                                logger.debug(context,
+                                    'Lifetime unregistration for device [%s] ended successfully',
+                                    obj.name);
+                            }
+                        });
+                    }
+                });
+            }
+        });
+    }, lifetimeCheckInterval);
+}
+
+/**
+ * Stops checking for device lifetime.
+ */
+function stopLifetimeCheck() {
+    clearInterval(checkLifetimeInterval);
+    checkLifetimeInterval = null;
+}
+
 /**
  * Initializes the device registry based on the parameter found in the configuration. For this in memory registry this
  * function doesn't do anything.
  *
- * @param {Object} config           Configuration object.
+ * @param {Object} newConfig           Configuration object.
+ * @param {Object} newContext          Context object.
+ * @param {Object} newLogger           Logger object.
  */
-function init(config, callback) {
+function init(newConfig, newContext, newLogger, callback) {
```

Comment:
Just a last detail. The logger and the context should not be passed from one module to the other. The logger can be required as any other module, and the context is meant to add information specific to the module where it is being used. In this case, please, define a `context` variable like this:
```
context = {
        op: 'LWM2MLib.MemoryRegistry'
    }
```
or `'LWM2MLib.MemoryRegistry'` in the case of the MongoDB registry, and do not pass this parameters.